### PR TITLE
[SRP-23 ] Setrf and trim actions

### PR DESCRIPTION
**Pull request by navneet-tw**
Objective: migration kafka topics to aws completely and achieve replication factor of 3.

the actions used are 
1. clone - migrate leader and create new replica in aws if needed 
2. setrf - create new replica to an aws broker (where not already present) if current replication factor is less than target
3. trim - remove a replica from a lw broker

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,64 +1,101 @@
...
+set-replication-factor: This action expects the list of aws brokers, set of topics and a target replication factor and creates a new replica if current replication factor is less than target.
+
+trim: This action expects the list of lw brokers and set of topics and reduce one replica of topic partition on lw brokers if possible (not possible when no replicas present on lw)
+
+Important note: The actions will only make any real changes to cluster when the kafka-assigner command (the main command for all actions) has --execute flag
+
+Running the actions (staging cluster)
+-------------------------------------
+
+Note: We will provide a driver script to run these actions automatically and the script will only expect the input as target replication factor and list of topics
```
Comment (sszp) [inline README.rst:45]: We do not specify the target replication factor, do we?

File: kafka/tools/models/partition.py [hunk 2]
```diff
@@ -72,6 +72,34 @@ def remove_replica(self, broker):
...
+    def remove_lw_replica(self, lw_broker_replicas):
+        # IMP: remove the last element as broker.partitions indexing scheme could break otherwise
+        if lw_broker_replicas:
+            self._remove_lw_replica(lw_broker_replicas[-1])
+        else:
+            print("no lw replicas for partition num {0} of topic {1} left to be removed".format(self.num, self.topic.name))
+
+    # remove lw replica on input broker (assume lw_broker has a replica for this partition)
+    def _remove_lw_replica(self, lw_broker):
+        log.info("removing lw replica now")
```
Comment (sszp) [inline kafka/tools/models/partition.py:88]: please add broker id here

**Review by sszp — APPROVED** [review]

File: README.rst (hunk 1 above)
Comment (navneet-tw) [reply README.rst:45]: till this pr we do, plan to make it optional and keep a default of 3

File: kafka/tools/models/partition.py (hunk 2 above)
Comment (navneet-tw) [reply kafka/tools/models/partition.py:88]: its' added here https://github.com/transferwise/kafka-tools/pull/2/files#diff-bfaeae8c8748c0da22398b6b24ff9e6bR97

**Comment by navneet-tw** [comment]
setrf and clone actions

**Pull request merged**